### PR TITLE
Outline the structure of the documentation

### LINKED ISSUE #10: Documentation Outline
Sketch out the basic outline for the documentation. First task to getting documentation written.

### PR DESCRIPTION
**Pull request by noelwelsh**
This PR ended up being bigger than I expected, as I ran into a problem with the doc tool (`mdoc`) that required updating the build. So there is a lot of build related stuff in there as well.

I'll finish up the quick start when I have some changes merged to `Plot` to make it easier to use.

Closes #10

### CODE REVIEW

**Comment by danielost** [comment]
There occurred a conflict in `Plot` after I merged PR https://github.com/creativescala/chartreuse/pull/8. Should I resolve it or you will do it?

**Comment by noelwelsh** [comment]
I think I've resolved it.

**Comment by noelwelsh** [comment]
Alright, I think it's all working.

File: docs/src/pages/development.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+There are an infinite number of possible `Layouts`, so we cannot enumerate them all, but we don't need to so long as the interface is consistent.
+Interfaces are represented as Scala `traits`.
+
+
+## Builders
+
+Components that have extensive options, but are limited in the number of possible implementations, are represented as builders.
+`Layer` is an example.
+`Plot` is an example.
+It acts a container fro all the other elements of a visualization.
```
Comment (danielost) [inline docs/src/pages/development.md]: A small typo here - `fro` (line 33)

**Comment by danielost** [comment]
I run into the following error when executing the `build` task:

```bash
[error] (docs / css) java.io.IOException: Cannot run program "npx": CreateProcess error=2, The system cannot find the file specified
```

But there is a big chance it's not associated with your changes, as the error first occurred yesterday. It might be my local problem. I just want to make sure I can omit it while reviewing the PR.

**Review by danielost — CHANGES_REQUESTED** [review]

**Comment by noelwelsh** [comment]
> I run into the following error when executing the `build` task:
> 
> ```shell
> [error] (docs / css) java.io.IOException: Cannot run program "npx": CreateProcess error=2, The system cannot find the file specified
> ```
> 
> But there is a big chance it's not associated with your changes, as the error first occurred yesterday. It might be my local problem. I just want to make sure I can omit it while reviewing the PR.

This is related to `npm`, which is Javascript, which is used to build the CSS for the documentation. You can install `npm` yourself or ignore it if you don't want to work locally on the documentation.

File: docs/src/pages/development.md (hunk 1 above)
Comment (noelwelsh) [reply docs/src/pages/development.md]: :+1:  Fixed.

**Comment by danielost** [comment]
> This is related to `npm`, which is Javascript, which is used to build the CSS for the documentation. You can install `npm` yourself or ignore it if you don't want to work locally on the documentation.

The problem is that it's installed and I tried reinstalling it. But still gives an error. I'll try to fix it somehow else.

**Review by danielost — APPROVED** [review]

**Comment by danielost** [comment]
The documentation is great!

**Pull request merged**